[next/neo] update to neogradle 7.0.43

Cheos137 opened the pull request:
and thus switching from mcp to neoform

Cheos137 committed:
update to neogradle 7.0.43

thus, switching from mcp to neoform

Cheos137 committed:
add foojay resolver convention to fix ci

Cheos137 committed:
configure test case to use foojay resolver convention

Cheos137 committed:
bump trainingwheels from 1.0.17 to 1.0.32

Cheos137 committed:
fix testcase settings.gradle creation

Cheos137 committed:
apply java plugin as a dummy in testcase

as "settingsPlugins" aren't generated by trainingwheels if "plugins" is empty (bug on their end)
additionally, use #withToolchains to abbreviate the foojay resolver convention specification

Cheos137 committed:
fix wrong download task being used

originally tried replacing the (now removed from neoform) DownloadArtifact.class with DownloadCore.class
however, the correct replacement would be Download.class with some additional configuration

XFactHD commented on src/main/java/org/parchmentmc/librarian/neogradle/ParchmentMappingVersion.java:
```diff
@@ -67,14 +67,14 @@ public String mcVersion() {
     @Nonnull
-    public String mcpVersion() {
-        return mcpVersion;
+    public String mneoformVersion() {
```
You got a sneaky typo here, this should be `neoformVersion()` without a leading `m`.

Cheos137 replied:
thanks, am actually surprised i didn't catch that one earlier

Cheos137 committed:
fix typo

sciwhiz12 commented:
@ParchmentMC/external-tools, should we permanently move Librarian to target NeoGradle (as the base branch of this PR does), or should we keep them co-existing, such that we have Librarian-FG and Librarian-NG?

I personally think we should go for the latter, in the interests of those moddings on versions where their preferred choice is Forge (and ForgeGradle) and of keeping the ability to update Librarian for those versions if we do more changes in the future.

marchermans commented:
In practice I want this code to live in NeoGradle anyway
And not in Librarian.

Cheos137 committed:
bump trainingwheels to 1.0.39

as of neoforged using github actions, trainingwheels is now published to maven central under a different groupId and artifactId

additionally, this allowes us to remove the workaround previously introduced in 83f5253d as it's now fixed

Cheos137 committed:
pin groovy_version to 3.0

as spock-core is only available to us using the maj.min scheme (instead of maj.min.fix) for groovy versions now

Cheos137 committed:
bump neogradle to 7.0.47

Cheos137 committed:
use neoforge 1.20.2 in test case

as minecraftforge is no longer available in the neoforged repository

marchermans commented:
This has been implemented natively in NG7+
